beacon-chain/rpc/eth/beacon/handlers.go

```diff
@@ -631,3 +637,120 @@ func (bs *Server) validateEquivocation(blk interfaces.ReadOnlyBeaconBlock) error
 	}
 	return nil
 }
+
+// GetBlockRoot retrieves hashTreeRoot of ReadOnlyBeaconBlock/BeaconBlockHeader.
+func (bs *Server) GetBlockRoot(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "beacon.GetBlockRoot")
+	defer span.End()
+
+	afterPath, ok := strings.CutPrefix(r.URL.Path, "/eth/v1/beacon/blocks/")
+	if !ok {
+		http2.HandleError(w, "Invalid path", http.StatusBadRequest)
+		return
+	}
+	blockID, ok := strings.CutSuffix(afterPath, "/root")
+	if !ok {
+		http2.HandleError(w, "Invalid path", http.StatusBadRequest)
+		return
+	}
+
+	var err error
+	var root []byte
+	switch blockID {
+	case "head":
+		root, err = bs.ChainInfoFetcher.HeadRoot(ctx)
+		if err != nil {
+			http2.HandleError(w, "Could not retrieve head block: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if root == nil {
+			http2.HandleError(w, "No head root was found", http.StatusNotFound)
+			return
+		}
+	case "finalized":
+		finalized := bs.ChainInfoFetcher.FinalizedCheckpt()
+		root = finalized.Root
+	case "genesis":
+		blk, err := bs.BeaconDB.GenesisBlock(ctx)
+		if err != nil {
+			http2.HandleError(w, "Could not retrieve blocks for genesis slot: "+err.Error(), http.StatusInternalServerError)
+			return
+		}
+		if err := blocks.BeaconBlockIsNil(blk); err != nil {
+			http2.HandleError(w, "Could not find genesis block: "+err.Error(), http.StatusNotFound)
+			return
+		}
+		blkRoot, err := blk.Block().HashTreeRoot()
+		if err != nil {
+			http2.HandleError(w, "Could not hash genesis block", http.StatusInternalServerError)
+			return
+		}
+		root = blkRoot[:]
+	default:
+		blockIDBytes, err := hexutil.Decode(blockID)
```

Comment:
I think this is wrong if I'm reading 709 correctly... this appears that you are trying to decode a hex value but if it's the values of slot then it will fail. you will need to rethink how to differentiate slot and hex here.

Reply:
One way would be to write `isHex := strings.HexPrefix(blockID, "0x")` and then use `isHex` instead of `len(blockIDBytes == 32` in the if/else